compiler/shared_ast/optimizations.ml

```diff
@@ -207,8 +207,9 @@ let rec optimize_expr :
       in
       EMatch { e = arg; cases; name = n1 }
     | EApp { f = EAbs { binder; _ }, _; args }
-      when binder_vars_used_at_most_once binder ->
-      (* beta reduction when variables not used. *)
+      when binder_vars_used_at_most_once binder
```

Comment:
This supercedes the same thing I just did yesterday in the C backend :) https://github.com/CatalaLang/catala/pull/532/commits/91f845189907b921dbcb0a999008937b6aa19b1f